### PR TITLE
Fix untar for centrifuge

### PR DESCRIPTION
**Pull request by sofstam**
Update centrifuge module so that the database is extracted using the untar module

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by jfy133 — CHANGES_REQUESTED** [review]
Don't forget to update `meta.yml` to specify the database is now just a directory.

See my comment about the slightly ugly `-x` requirements by centrifuge and a possible solution to make it more flexible wihtin the nf-core module

File: modules/centrifuge/main.nf [hunk 1]
```diff
@@ -42,9 +41,8 @@ process CENTRIFUGE {
     }
     def sam_output = sam_format ? "--out-fmt 'sam'" : ''
     """
-    tar -xf $db
     centrifuge \\
-        -x $db_name \\
+        -x ${db}/${db} \\
```
[the remark this answers is not in the document]
Comment (sofstam) [reply modules/centrifuge/main.nf:45]: Thank you for your comments.I will update the `meta.yml`

I did not like the solution with `-x`  either but a second parameter sounds a good idea. If I understand you suggestion, that would mean:

```
tuple val(meta), path(reads)
path db
val db_name
val save_unaligned
val save_aligned
val sam_format
```

[the remark this answers is not in the document]
Comment (jfy133) [reply modules/centrifuge/main.nf:45]: Yup exactly

File: modules/centrifuge/meta.yml [hunk 2]
```diff
@@ -27,6 +27,9 @@ input:
       type: directory
       description: Centrifuge database in .tar.gz format
       pattern: "*.tar.gz"
+  - db_name:
+      type: string
+      description: Centrifuge filenames without the suffix ".cf"
```
Comment (jfy133) [inline modules/centrifuge/meta.yml:32]:
```suggestion
      description: Centrifuge database filenames without the suffix ".cf"
```

File: modules/centrifuge/main.nf [hunk 3]
```diff
@@ -42,9 +43,8 @@ process CENTRIFUGE {
     }
     def sam_output = sam_format ? "--out-fmt 'sam'" : ''
     """
-    tar -xf $db
     centrifuge \\
-        -x $db_name \\
+        -x ${db}/$db_name \\
```
Comment (jfy133) [inline modules/centrifuge/main.nf:47]:
```suggestion
        -x ${db}/${db_name} \\
```

File: tests/modules/centrifuge/main.nf [hunk 4]
```diff
@@ -2,18 +2,21 @@
...
+include { UNTAR      } from '../../../modules/untar/main.nf'
 include { CENTRIFUGE } from '../../../modules/centrifuge/main.nf'
 
 workflow test_centrifuge_single_end {
     input = [ [ id:'test', single_end:true ], // meta map
               [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
             ]
-    db   =  file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/minigut_cf.tar.gz", checkIfExists: true)
+    db    =  [ [], file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/minigut_cf.tar.gz', checkIfExists: true) ]
+    db_name = file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/minigut_cf.tar.gz').toString().replace(".tar.gz","")
```
Comment (jfy133) [inline tests/modules/centrifuge/main.nf:13]:
```suggestion
    db_name = "minigut_cf"
```

Comment (jfy133) [reply tests/modules/centrifuge/main.nf:13]: Don't need to stage twice for these tests :+1:

File: tests/modules/centrifuge/main.nf [hunk 5]
```diff
@@ -22,12 +25,14 @@ workflow test_centrifuge_paired_end {
               [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                 file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
             ]
-     db   =  file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/minigut_cf.tar.gz", checkIfExists: true)
+     db    =  [ [], file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/minigut_cf.tar.gz', checkIfExists: true) ]
+     db_name = file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/minigut_cf.tar.gz').toString().replace(".tar.gz","")
```
Comment (jfy133) [inline tests/modules/centrifuge/main.nf:29]:
```suggestion
     db_name = "minigut_cf"
```

**Review by jfy133 — APPROVED** [review]
Small suggestions to make tests slightly more smoother but otherwise I think we are good to go.

**Comment by sofstam** [comment]
I committed your suggestions, thanks for all the comments. Shall I merge now?

**Comment by jfy133** [comment]
Go ahead!

**Pull request merged**